metacat-connector-druid/src/main/java/com/netflix/metacat/connector/druid/DruidConnectorTableService.java

```diff
@@ -0,0 +1,106 @@
+import com.google.common.collect.Lists;
+import com.netflix.metacat.common.QualifiedName;
+import com.netflix.metacat.common.dto.Pageable;
+import com.netflix.metacat.common.dto.Sort;
+import com.netflix.metacat.common.server.connectors.ConnectorRequestContext;
+import com.netflix.metacat.common.server.connectors.ConnectorTableService;
+import com.netflix.metacat.common.server.connectors.ConnectorUtils;
+import com.netflix.metacat.common.server.connectors.exception.ConnectorException;
+import com.netflix.metacat.common.server.connectors.model.TableInfo;
+import com.netflix.metacat.connector.druid.converter.DataSource;
+import com.netflix.metacat.connector.druid.converter.DruidConnectorInfoConverter;
+import com.netflix.metacat.connector.druid.converter.DruidConverterUtil;
+import lombok.extern.slf4j.Slf4j;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Comparator;
+import java.util.List;
+
+/**
+ * Druid Connector Table Service.
+ *
+ * @author zhenl
+ * @since 1.1.0
+ */
+@Slf4j
+public class DruidConnectorTableService implements ConnectorTableService {
+    private final MetacatDruidClient druidClient;
+    private final DruidConnectorInfoConverter druidConnectorInfoConverter;
+    /**
+     * Constructor.
+     *
+     * @param druidClient druid client
+     * @param druidConnectorInfoConverter druid infor object converter
+     */
+    public DruidConnectorTableService(
+        final MetacatDruidClient druidClient,
+        final DruidConnectorInfoConverter druidConnectorInfoConverter
+    ) {
+        this.druidClient = druidClient;
+        this.druidConnectorInfoConverter = druidConnectorInfoConverter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public TableInfo get(@Nonnull final ConnectorRequestContext context, @Nonnull final QualifiedName name) {
+        log.debug("Beginning to get table metadata for qualified name {} for request {}", name, context);
+        final ObjectNode node = this.druidClient.getLatestDataByName(name.getTableName());
+        final DataSource dataSource = DruidConverterUtil.getDatasourceFromLatestSegmentJsonObject(node);
+        return this.druidConnectorInfoConverter.getTableInfoFromDatasource(dataSource);
+    }
+
+    /**
+     * {@inheritDoc}.
+     */
+    @Override
+    public List<QualifiedName> listNames(
+        final ConnectorRequestContext requestContext,
+        final QualifiedName name,
+        @Nullable final QualifiedName prefix,
+        @Nullable final Sort sort,
+        @Nullable final Pageable pageable
+    ) {
+        try {
+            final List<QualifiedName> qualifiedNames = Lists.newArrayList();
+            final String tableFilter = (prefix != null && prefix.isTableDefinition()) ? prefix.getTableName() : null;
+            for (String tableName : this.druidClient.getAllDataSources()) {
+                if (tableFilter == null || tableName.startsWith(tableFilter)) {
```

Comment:
This condition looks weird. First you enter the` if` condition when the table filter starts with but then again within the `if` block you again check to see if the prefix matches? That seems redundant. Seems like this should be an `if` followed by an `else if` but I might be reading it wrong.